File: Practice(Beginner)/C++/Cops and the Thief Devu.cpp
// Source: https://www.codechef.com/problems/COPS

#include <bits/stdc++.h>
using namespace std;

int main() {
    int t, m, x, y;
    cin >> t;
    while (t--){
        int arr[100] = {0};
        cin >> m >> x >> y;
        int police[m];
        for (int i = 0; i < m; ++i) {
            cin >> police[i];
        }
        int it = 0;
        while (it < m){
            int start = police[it] - (x*y) - 1;
            int finish = police[it] + (x*y);
            if(start < 0){
                start = 0;
            }
            if(finish > 100){
                finish = 100;
            }
            for (int i = start; i < finish; ++i) {
                arr[i] = 1;
            }
            it++;
        }
        cout << count(arr,arr+100,0) << endl;
    }
    return 0;
}

File: Practice(Beginner)/Python3/Cops_and_the_Thief_Devu.py
# Source: https://www.codechef.com/problems/COPS

t = int(input())
for test in range(t):
    l = [0]*100
    m, x, y = map(int, input().split())
    police = list(map(int, input().split()))
    for i in range(m):
        start = police[i] - (x*y) - 1
        finish = police[i] + (x*y)
        if(start < 0):
            start = 0
        if(finish > 100):
            finish = 100
        for j in range(start,finish):
            l[j] = 1;
    print(l.count(0))
